src/TextAnalyzer/Form.php

```diff
@@ -0,0 +1,220 @@
+     * @return array
+     */
+    public function getOptions(): array
+    {
+        return $this->options;
+    }
+
+    protected function applyDefaultOptions(): void
+    {
+        $defaultOptions = [
+            'method' => 'POST',
+            'action' => 'index.php',
+            'textarea_name' => 'textarea',
+            'encoding' => 'UTF-8',
+        ];
+
+        // Simple realization
+        $allowedEncoding = TextProcessor::ALLOWED_ENCODING;
+
+        // 'output' key in options array will be use to Output the result for task, just filter enter parameters
+        $this->options = array_diff(array_merge($defaultOptions, $this->options), [self::OUTPUT]);
+
+        // filter allowed encoding
+        $this->options['encoding'] = in_array($this->options['encoding'], $allowedEncoding) ? $this->options['encoding'] : 'UTF-8';
+    }
+
+    /**
+     * @param string $value
+     */
+    protected function setOutput(string $value): void
+    {
+        $this->options[self::OUTPUT] = htmlspecialchars($value);
+    }
+
+    /**
+     * @param string $input
+     * @return string
+     */
+    protected function prepareOutput(string $input): string
+    {
+        $textProcessor = new TextProcessor($this->options['encoding']);
+
+        // Start time
+        $startTime = hrtime(true);
+
+        $output = <<< EOT
+Number of characters: {$textProcessor->calculateLength($input)}
+Number of words: {$textProcessor->calculateWords($input)}
+Number of sentences: {$textProcessor->calculateSentences($input)}
+Frequency of characters: {$textProcessor->calculateFrequencyCharacters($input)}
+Distribution of characters as a percentage of total: {$textProcessor->calculateCharactersDistributions($input)}
+Average word length: {$textProcessor->calculateAverageWordLength($input)}
+The average number of words in a sentence: {$textProcessor->calculateAverageNumberOfWords($input)}
+Top 10 most used words: {$textProcessor->calculateMostUsageWords($input)}
+Top 10 longest words: {$textProcessor->getLongestWords($input)}
+Top 10 shortest words: {$textProcessor->getShortestWords($input)}
+Top 10 longest sentences: {$textProcessor->getLongestSentences($input)}
+Top 10 shortest sentences: {$textProcessor->getShortestSentences($input)}
+Number of palindrome words: {$textProcessor->calculatePalindromeWords($input)}
+Top 10 longest palindrome words: {$textProcessor->getLongestPalindromesWords($input)}
+Is the whole text a palindrome? (Without whitespaces and punctuation marks.) {$textProcessor->isWholeTextPalindromes($input)}
+
+EOT;
+
+        $endTime = hrtime(true);
+        $timeSpendINms = ($endTime - $startTime) * 1e-6;
+
+        $output .= "The time it took to process the text in ms: $timeSpendINms";
+
+        return $output;
+    }
+
+    /**
+     * @param string $form
+     * @return string
+     * @throws InvalidArgumentException
+     */
+    protected function validateForm(string $form = ''): string
+    {
+        if (strlen($form) === 0) {
```

Comment:
I understand that you wanted to predict all possible cases and make the code bugless, but in my opinion that`s kind of "overengeneering".